lib/cocoapods-core/podfile/target_definition.rb

```diff
@@ -921,7 +921,8 @@ def parse_subspecs(name, requirements)
         end if subspecs
 
         test_specs.each do |ss|
-          store_pod("#{name}/#{ss}", *requirements.dup)
+          requirements_copy = requirements.map(&:dup)
+          store_pod("#{name}/#{ss}", *requirements_copy)
```

Comment:
this was basically returning a shallow clone of the requirements and when using test specs we are actually *adding* a pod not overriding like it happens for when using `:subspecs`.

The shallow clone caused the `store_pod` to remove the configuration when parsing the test spec and then the parent had 0 configuration causing an inconsistency between the parent configurations and the test spec configuration.